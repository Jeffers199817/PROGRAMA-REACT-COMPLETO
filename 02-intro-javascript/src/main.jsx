//import { heroes } from './data/heroes'
//import { heroes} from './data/heroes'

import { heroes } from "./data/heroes";


const getHeroById = (id) => {
  
  return heroes.find((idHeroes)=> idHeroes.id==id);
}

console.log(getHeroById(5));