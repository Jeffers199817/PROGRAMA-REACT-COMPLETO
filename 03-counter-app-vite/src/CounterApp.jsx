
import PropTypes from "prop-types"
import { useState } from "react";



export const CounterApp = ({ value = 4 }) => {

    const [counter, setCounter] = useState(value);

    //FUNCIONES
    //AGREGAR
    const handleAdd = () => { setCounter((c) => c + 1); };
    //SUBTRACT
    const handleSubtract = () => {
        if (counter > 0) {
            setCounter((c) => c - 1);
        }
    }
        
        //RESET
        const handleReset = () => { setCounter((c) => (c * 0)) }


    
        return <>
    
            <h1>CounterAPP</h1>
            <h2>{counter}</h2>
            <button onClick={handleAdd}> +1</button >
            <button onClick={handleSubtract}> --1</button>
            <button onClick={handleReset}>Reset</button>
    
        </>


    }


    CounterApp.propTypes = {
        value: PropTypes.number.isRequired,
    }

