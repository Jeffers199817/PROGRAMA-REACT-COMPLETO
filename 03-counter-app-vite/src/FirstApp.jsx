import PropTypes from "prop-types";

export const FirstApp = ({title = "Hola, soy Vegeta" , subtitulo="No hay subtitulo", name="Jefferson"}) => {
  return (
    <>
      <h1 data-testid ='test-title'>{title} </h1>
      <h2>{subtitulo}</h2>
    
      <h3>{name}</h3>
      
    </>
  );
};

FirstApp.propTypes = {
  //title: PropTypes.string.isRequired,
 
};
