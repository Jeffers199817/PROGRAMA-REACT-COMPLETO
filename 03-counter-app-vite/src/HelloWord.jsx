
export const HelloWord = () => {
    return <h1>Hello word</h1>
}

export const FirstApp= () => {
    return <h1>FirtApp</h1>
}