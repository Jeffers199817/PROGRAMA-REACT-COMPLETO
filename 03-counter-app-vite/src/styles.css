

html, body{

    background-color: #212324;
    color:white;
    font-family:Helvetica,, sans-serif;
    font-size:1.3rem;
    padding: 70px;
}


p{
    color: red;
    font-family:

}