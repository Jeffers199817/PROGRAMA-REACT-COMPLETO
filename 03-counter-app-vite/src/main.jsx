import React from "react";
import ReactDOM from "react-dom/client";
import { FirstApp } from "./FirstApp";
import { CounterApp } from './CounterApp';
import "./styles.css";

ReactDOM.createRoot(document.getElementById("root")).render(
    <React.StrictMode>
        <FirstApp title="Hola, soy Vegeta"/> {/* Omite title y subTitle para probar defaultProps */}
  </React.StrictMode>
);
