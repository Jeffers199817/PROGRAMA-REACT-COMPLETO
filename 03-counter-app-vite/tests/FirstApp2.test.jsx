import { render, screen } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";



describe('Pruebas en <FirstApp.jsx></FirstApp.jsx>', () => {


    const title = "Hola soy goku.";
    const subtitle = "Esto es un subtitulo";

    test('Debe de hacer match con el snapshot', () => { 

       
        const {container} = render(<FirstApp title={title} />)
       
        expect(container).toMatchSnapshot();

    })
    

    test('Debe mostrar el mensaje hola soy goku', () => {
        
        render(<FirstApp title={title} />)
   

        
        expect(screen.getByText(title)).toBeTruthy();
    })

    test('Debe mostrar el titulo en un h1', () => { 

        render(<FirstApp title={title} />)
       expect(screen.getByRole('heading',{level:1}).innerHTML).toContain(title);
    })
    
    test('debe de mostar el subtitulo enviado por props', () => { 

        render(<FirstApp subtitulo={subtitle} />)

        expect(screen.getByRole('heading',{level:2}).innerHTML).toContain(subtitle);

    
    })
    
    
 })