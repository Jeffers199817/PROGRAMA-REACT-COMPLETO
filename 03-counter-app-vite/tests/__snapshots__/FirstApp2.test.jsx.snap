// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp.jsx></FirstApp.jsx> Debe de hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola soy goku.
     
  </h1>
  <h2>
    No hay subtitulo
  </h2>
  <h3>
    Jefferson
  </h3>
</div>
`;
